### PR TITLE
Added reconciler test and updated deps

### PR DESCRIPTION
**Pull request by jihuin**
## Proposed Changes

* Added reconciler test based on https://github.com/knative/serving/blob/master/pkg/reconciler/reconciler_test.go
* Updated deps by `dep ensure` and `./hack/update-deps.sh`

**Release Note**

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/reconciler_test.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+*/
+
+package reconciler
+
+import (
+	"context"
+	"testing"
+
+	_ "knative.dev/pkg/client/injection/client/fake"
+    _ "knative.dev/pkg/client/injection/kube/client/fake"
```
Comment (trshafer) [inline pkg/reconciler/reconciler_test.go]: nit: spacing

Comment (jihuin) [reply pkg/reconciler/reconciler_test.go]: resolved

**Comment by anniefu** [comment]
It might be worth minimizing the change to only the package needed for this test instead of updating all of them. i.e. "dep ensure -update knative.dev/pkg"

Running ./update-deps.sh to update all of vendor/ should probably be isolated to its own PR to keep updating misc. packages and adding new feature/test code in independent commits that can be reverted without impact each other.

**Comment by jihuin** [comment]
@anniefu  The changes are all about adding fake clients that are required for the test.

**Pull request merged**